### ISSUE TITLE
Connecting to open WiFis fails

### ISSUE THREAD

**Issue by mzanetti** [nymea/nymea-networkmanager#22]
See https://github.com/nymea/berrylan/issues/6 for more info.

**Comment by maxpage92** [comment]
I was just wondering if this issue being closed means that it is fixed? I came her from the berrylan side of things. If it isn’t, I’d be happy to help the dev team where i can :)

**Comment by mzanetti** [comment]
It requires 2 fixes. One in the BerryLan app, the other in here. The one in BerryLan has been fixed, but in nymea-networkmanager it's still open.

**Comment by maxpage92** [comment]
Yeah makes sense. The nymea networkmanager needs to enter it differently if there is no password compared to just an empty one. WPA supplicant is the same. Can I help? I don’t have experience in the nymea code, but with some digging and experimenting on how it works i’m sure i could be of use :) or do you guys have a fix planed? Then I’ll just stay out of it haha

**Comment by mzanetti** [comment]
Feel free to try and get it fixed, we'd appreciate it. I just had a quick look to see if I could give you any pointers and I think I'd probably start here:

https://github.com/nymea/libnymea-networkmanager/blob/master/libnymea-networkmanager/networkmanager.cpp#L127

You're welcome to open a pull request with fixes.

**Comment by maxpage92** [comment]
I’ll have a look, if i see anything i’ll have a go and report.

**Comment by maxpage92** [comment]
How did you fix it in berrylan? Does it just pass a password variable that is empty, or do you not pass that variable anymore?

**Comment by mzanetti** [comment]
It'll be empty

**Comment by t-mon** [comment]
Good news, I fixed this issue yesterday and open WiFi networks are supported now. 
There was a bug in the verification if an access point is protected or not. This has been fixed and the content of the password variable will be ignored, since the server already knows if the access point is protected or not. 
Once the current version is released (probably within the next week), you just have to update and it should work out of the box.

**Comment by maxpage92** [comment]
Fantastic! I tried playing with the code, but didn’t spend enough time to figure it out... quite glad i didn’t now :) When do you expect the update to the network manager to be merged/available to add to our devices?
